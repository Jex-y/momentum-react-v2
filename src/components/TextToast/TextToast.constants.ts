import { TextToastTextAlign } from './TextToast.types';

const CLASS_PREFIX = 'md-text-toast';

const DEFAULTS = {
  ALIGNMENT: 'center' as TextToastTextAlign,
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
  text: `${CLASS_PREFIX}-text`,
};

const ALIGNMENT: Record<string, TextToastTextAlign> = {
  LEFT: 'left',
  CENTER: 'center',
};

export { CLASS_PREFIX, DEFAULTS, STYLE, ALIGNMENT };
