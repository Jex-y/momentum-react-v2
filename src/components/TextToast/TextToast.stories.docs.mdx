The `<TextToast />` component. This component is designed to retrieve props like the `text` to display and the `textAlignment`, which can be `center` or `left` at the moment.
If no or an empty `text` is passed in, the component will not be rendered. If no `textAlignment` is passed in, the default alignment will be `center`.
