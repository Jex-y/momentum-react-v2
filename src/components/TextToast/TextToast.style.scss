.md-text-toast-wrapper {
  display: flex;
  padding: 0.5rem 0.75rem;
  width: 20rem;
  border: 0.0625rem solid var(--textToast-border-color);
  border-radius: 0.5rem;
  background-color: var(--textToast-background);
  box-shadow: var(--md-globals-elevation-2);
  align-items: center;
  box-sizing: content-box !important;

  &[data-text-alignment='center'] {
    justify-content: center;
    text-align: center;
  }

  &[data-text-alignment='left'] {
    justify-content: flex-start;
    text-align: left;
  }

  > .md-icon-wrapper {
    margin-right: 0.3125rem;
  }
}

.md-text-toast-text {
  color: var(--textToast-text);
}
