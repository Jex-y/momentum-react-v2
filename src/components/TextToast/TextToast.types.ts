import { CSSProperties } from 'react';
import { IconProps } from '../Icon';

export type TextToastTextAlign = 'left' | 'center';

export interface Props {
  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;

  /**
   * Text to be displayed in TextToast component
   */
  text: string;

  /**
   * Alignment of the Text in the TextToast component.
   *
   * @default center
   */
  textAlignment?: TextToastTextAlign;

  /**
   * IconProps to be passed in - allows defining the icon which should
   * be shown on the left-hand side of the text. At least the `name` property
   * should be passed in through the object, otherwise the Icon will not be shown.
   *
   * @default undefined
   */
  iconProps?: IconProps;
}
