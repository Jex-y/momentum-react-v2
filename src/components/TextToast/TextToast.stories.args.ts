import { commonStyles } from 'storybook/helper.stories.argtypes';

import { ALIGNMENTS } from './TextToast.constants';

const textToastArgTypes = {
  text: {
    description: 'Text to be displayed in TextToast component',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  textAlignment: {
    description: 'Alignment of the Text in the TextToast component.',
    control: { type: 'select', options: [undefined, ...Object.values(ALIGNMENTS)] },
    table: {
      type: {
        summary: Object.values(ALIGNMENTS)
          .map((value) => `"${value}"`)
          .join(' | '),
      },
      defaultValue: {
        summary: 'center',
      },
    },
  },
  iconProps: {
    description: `IconProps to be passed in - allows defining the icon which should
    be shown on the left-hand side of the text. At least the \`name\` property
    should be passed in through the object, otherwise the Icon will not be shown.`,
    control: { type: 'object' },
    table: {
      type: {
        summary: 'object',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
};

export { textToastArgTypes };

export default {
  ...commonStyles,
  ...textToastArgTypes,
};
