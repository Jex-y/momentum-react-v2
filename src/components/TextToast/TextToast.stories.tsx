import StyleDocs from 'storybook/docs.stories.style.mdx';
import { DocumentationPage } from 'storybook/helper.stories.docs';
import { MultiTemplate, Template } from 'storybook/helper.stories.templates';

import argTypes from './TextToast.stories.args';
import Documentation from './TextToast.stories.docs.mdx';

import TextToast, { TextToastProps } from './';

export default {
  title: 'Momentum UI/TextToast',
  component: TextToast,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
};

const Example = Template<TextToastProps>(TextToast).bind({});

Example.argTypes = { ...argTypes };

Example.args = {
  text: 'Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.',
  textAlignment: 'center',
  iconProps: undefined,
};

const Common = MultiTemplate<TextToastProps>(TextToast).bind({});

Common.argTypes = { ...argTypes };
delete Common.argTypes.children;

Common.args = {
  text: 'Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.',
  textAlignment: 'center',
  iconProps: undefined,
};

Common.parameters = {
  variants: [
    { textAlignment: 'left' },
    { textAlignment: 'center' },
    {
      text: "You're temporarily unmuted",
      textAlignment: 'center',
      iconProps: {
        name: 'sound-default',
        fillColor: 'var(--banner-success-text)',
        autoScale: true,
      },
      style: { width: 'fit-content' },
    },
  ],
};

export { Example, Common };
