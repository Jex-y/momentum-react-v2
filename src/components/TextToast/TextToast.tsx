import React, { FC } from 'react';
import classnames from 'classnames';

import { DEFAULTS, STYLE } from './TextToast.constants';
import { Props } from './TextToast.types';
import './TextToast.style.scss';
import Text from '../Text/Text';

/**
 * The TextToast component renders a provided text.
 */
const TextToast: FC<Props> = (props: Props) => {
  const { text, textAlignment = DEFAULTS.ALIGNMENT, className, id, style } = props;

  return (
    <>
      {text?.length > 0 && (
        <div className={classnames(className, STYLE.wrapper)} id={id} style={style}>
          <Text
            type="body-secondary"
            className={classnames(className, STYLE.text)}
            style={{ textAlign: textAlignment }}
          >
            {text}
          </Text>
        </div>
      )}
    </>
  );
};

export default TextToast;
