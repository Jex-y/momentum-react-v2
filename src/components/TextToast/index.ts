import { default as TextToast } from './TextToast';
import * as CONSTANTS from './TextToast.constants';
import { Props } from './TextToast.types';

export { CONSTANTS as TEXT_TOAST_CONSTANTS };

export type TextToastProps = Props;

export default TextToast;
