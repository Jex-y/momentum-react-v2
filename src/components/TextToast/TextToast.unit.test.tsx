import React from 'react';
import { mount } from 'enzyme';

import TextToast, { TEXT_TOAST_CONSTANTS as CONSTANTS } from './';
import Text from '../Text/Text';

const textMock =
  'Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.';

describe('<TextToast />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<TextToast text={textMock} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(<TextToast text={textMock} className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(<TextToast text={textMock} id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(<TextToast text={textMock} style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with textAlignment `left`', () => {
      expect.assertions(1);

      const container = mount(<TextToast text={textMock} textAlignment="left" />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(<TextToast text={textMock} />)
        .find(TextToast)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = mount(<TextToast text={textMock} className={className} />)
        .find(TextToast)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = mount(<TextToast text={textMock} id={id} />)
        .find(TextToast)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = mount(<TextToast text={textMock} style={style} />)
        .find(TextToast)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should not render when an empty text is provided', () => {
      expect.assertions(1);

      const element = mount(<TextToast text="" />)
        .find(TextToast)
        .getDOMNode();

      expect(element).toBeNull();
    });

    it('should have provided text when text is provided', () => {
      expect.assertions(1);

      const element = mount(<TextToast text={textMock} />)
        .find(Text)
        .getDOMNode();

      expect(element.textContent).toBe(textMock);
    });

    it('should have default text alignment `center` when no textAlignment is provided', () => {
      expect.assertions(1);

      const element = mount(<TextToast text={textMock} />)
        .find(Text)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe('text-align: center;');
    });

    it('should have provided text alignment when textAlignment is provided', () => {
      expect.assertions(1);

      const element = mount(<TextToast text={textMock} textAlignment="left" />)
        .find(Text)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe('text-align: left;');
    });
  });
});
