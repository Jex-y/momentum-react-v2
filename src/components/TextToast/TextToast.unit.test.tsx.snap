// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextToast /> snapshot should match snapshot 1`] = `
<TextToast
  text="Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte."
>
  <div
    className="md-text-toast-wrapper"
  >
    <Text
      className="md-text-toast-text"
      style={
        Object {
          "textAlign": "center",
        }
      }
      type="body-secondary"
    >
      <small
        className="md-text-wrapper md-text-toast-text"
        data-type="body-secondary"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.
      </small>
    </Text>
  </div>
</TextToast>
`;

exports[`<TextToast /> snapshot should match snapshot with className 1`] = `
<TextToast
  className="example-class"
  text="Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte."
>
  <div
    className="example-class md-text-toast-wrapper"
  >
    <Text
      className="example-class md-text-toast-text"
      style={
        Object {
          "textAlign": "center",
        }
      }
      type="body-secondary"
    >
      <small
        className="md-text-wrapper example-class md-text-toast-text"
        data-type="body-secondary"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.
      </small>
    </Text>
  </div>
</TextToast>
`;

exports[`<TextToast /> snapshot should match snapshot with id 1`] = `
<TextToast
  id="example-id"
  text="Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte."
>
  <div
    className="md-text-toast-wrapper"
    id="example-id"
  >
    <Text
      className="md-text-toast-text"
      style={
        Object {
          "textAlign": "center",
        }
      }
      type="body-secondary"
    >
      <small
        className="md-text-wrapper md-text-toast-text"
        data-type="body-secondary"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.
      </small>
    </Text>
  </div>
</TextToast>
`;

exports[`<TextToast /> snapshot should match snapshot with style 1`] = `
<TextToast
  style={
    Object {
      "color": "pink",
    }
  }
  text="Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte."
>
  <div
    className="md-text-toast-wrapper"
    style={
      Object {
        "color": "pink",
      }
    }
  >
    <Text
      className="md-text-toast-text"
      style={
        Object {
          "textAlign": "center",
        }
      }
      type="body-secondary"
    >
      <small
        className="md-text-wrapper md-text-toast-text"
        data-type="body-secondary"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.
      </small>
    </Text>
  </div>
</TextToast>
`;

exports[`<TextToast /> snapshot should match snapshot with textAlignment \`left\` 1`] = `
<TextToast
  text="Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte."
  textAlignment="left"
>
  <div
    className="md-text-toast-wrapper"
  >
    <Text
      className="md-text-toast-text"
      style={
        Object {
          "textAlign": "left",
        }
      }
      type="body-secondary"
    >
      <small
        className="md-text-wrapper md-text-toast-text"
        data-type="body-secondary"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Lorem ipsum dolor site aw aetns ctetuer adipiscing elit nullam amarte adipiscing elit nullam amarte.
      </small>
    </Text>
  </div>
</TextToast>
`;
